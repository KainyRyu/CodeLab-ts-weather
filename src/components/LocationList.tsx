import React from 'react';
import Styled from 'styled-components';
import { useGlobalContext } from 'context/GlobalContext';

const Container = Styled.div`
  width: 300px;
`;

const Row = Styled.div`
  display: flex;
  justify-content: space-between;
  border-bottom: white 1px solid;
  margin-top: 5px;
`;

export default function LocationList() {
  const { getCities } = useGlobalContext();
  const cities = JSON.parse(getCities);
  console.log(cities);
  return (
    <Container>
      {cities?.map((city: string) => (
        <Row key={city}>
          <div>{city}</div>
          <button>X</button>
        </Row>
      ))}
    </Container>
  );
}
