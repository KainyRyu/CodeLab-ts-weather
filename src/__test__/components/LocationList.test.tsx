import React from 'React';
import { render, screen } from '@testing-library/react';
import LocationList from 'components/LocationList';

describe('<LocationList/>', () => {
  it('renders component correctly', () => {
    const { container } = render(<LocationList />);

    const berlin = screen.getByText('berlin');
    expect(berlin).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
