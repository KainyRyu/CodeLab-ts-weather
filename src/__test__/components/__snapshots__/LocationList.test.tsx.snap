// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationList/> adds city 1`] = `
<div>
  <div
    class="sc-bdvvtL cgThLl"
  >
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="london"
      >
        london
      </div>
      <button
        data-testid="london"
        name="london"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="berlin"
      >
        berlin
      </div>
      <button
        data-testid="berlin"
        name="berlin"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="seoul"
      >
        seoul
      </div>
      <button
        data-testid="seoul"
        name="seoul"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="prague"
      >
        prague
      </div>
      <button
        data-testid="prague"
        name="prague"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="hongkong"
      >
        hongkong
      </div>
      <button
        data-testid="hongkong"
        name="hongkong"
      >
        X
      </button>
    </div>
    <input
      placeholder="add new location"
      type="text"
      value="hongkong"
    />
    <button>
      add
    </button>
  </div>
</div>
`;

exports[`<LocationList/> deletes city 1`] = `
<div>
  <div
    class="sc-bdvvtL cgThLl"
  >
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="london"
      >
        london
      </div>
      <button
        data-testid="london"
        name="london"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="berlin"
      >
        berlin
      </div>
      <button
        data-testid="berlin"
        name="berlin"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="seoul"
      >
        seoul
      </div>
      <button
        data-testid="seoul"
        name="seoul"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="prague"
      >
        prague
      </div>
      <button
        data-testid="prague"
        name="prague"
      >
        X
      </button>
    </div>
    <input
      placeholder="add new location"
      type="text"
      value=""
    />
    <button>
      add
    </button>
  </div>
</div>
`;

exports[`<LocationList/> gets location list from globalProvider 1`] = `
<div>
  <div
    class="sc-bdvvtL cgThLl"
  >
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="london"
      >
        london
      </div>
      <button
        data-testid="london"
        name="london"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="berlin"
      >
        berlin
      </div>
      <button
        data-testid="berlin"
        name="berlin"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="seoul"
      >
        seoul
      </div>
      <button
        data-testid="seoul"
        name="seoul"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="paris"
      >
        paris
      </div>
      <button
        data-testid="paris"
        name="paris"
      >
        X
      </button>
    </div>
    <div
      class="sc-gsDKAQ jAJQy"
    >
      <div
        data-name="prague"
      >
        prague
      </div>
      <button
        data-testid="prague"
        name="prague"
      >
        X
      </button>
    </div>
    <input
      placeholder="add new location"
      type="text"
      value=""
    />
    <button>
      add
    </button>
  </div>
</div>
`;

exports[`<LocationList/> renders component correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL cgThLl"
  >
    <input
      placeholder="add new location"
      type="text"
      value=""
    />
    <button>
      add
    </button>
  </div>
</div>
`;
